libgnucash/engine/Transaction.c

```diff
@@ -2472,15 +2481,18 @@ xaccTransGetTxnType (const Transaction *trans)
 {
     const char *s = NULL;
     GValue v = G_VALUE_INIT;
+    char ret = TXN_TYPE_NONE;
 
     if (!trans) return TXN_TYPE_NONE;
     qof_instance_get_kvp (QOF_INSTANCE (trans), &v, 1, TRANS_TXN_TYPE_KVP);
     if (G_VALUE_HOLDS_STRING (&v))
+    {
          s = g_value_get_string (&v);
-    if (s && strlen (s) == 1)
-        return *s;
-
-    return TXN_TYPE_NONE;
+         if (s && (s[0] != '\0') && (s[1] == '\0'))
+             ret = s[0];
+    }
+    g_value_unset (&v);
```

Comment:
I *think*  'g_value_take_string' works the other way around. It would be used to set a GValue with the given string such that it takes ownership of it. It's not a method to extract the string (a glaring omission in the GValue specification in retrospect).

Reply:
https://developer.gnome.org/gobject/stable/gobject-Standard-Parameter-and-Value-Types.html#g-value-take-string

Reply:
Darn, you're right.